9주차 코드리뷰

kang-kibong opened the pull request:
안녕하세요 멘토님,

9주차 궁금한점을 README.md에 작성하였습니다.

Storybook 배포링크는 다음과 같습니다!

https://66e528a32564a3669b75354b-yinrtzeafe.chromatic.com/

이번주차도 잘 부탁드립니다, 감사합니다!

kang-kibong committed:
9주차 산출물  (#91) (#92)

* Update README.md

* Refactor/#47 6주차 코드리뷰 리팩토링 (#48)

* refactor: separate style and prop-related constants

* refactor: remove lambda function

* refactor: restructure SignUp and RecruitmentHeader components

* feat: separate Button and Text component of SignIn

* feat: add SignUpText component

* refactor: remove auth page's barrel file

* Feat/#42, #43 지원자 목록 페이지 및 팝업 구현 (#51)

* feat: 지원자 목록 페이지 구현

* feat: 지원자 목록 페이지 스토리북 생성

* refactor: ApplicantList 테이블 컴포넌트 분리

* feat: Applicants path 설정

* feat: 지원자 목록 페이지에 계약 관련 팝업 추가

* fix: 이미지 경로 수정 및 불필요한 태그 제거

* refactor: MyAccount 페이지 구조 변경 및 CompanyRecruitments로 파일명 변경

- 기존 MyAccount 페이지를 CompanyRecruitments로 이름 변경
- 새로운 MyAccount 페이지 구현을 위해 기존 페이지의 역할 변경

* refactor: visaRegistration 관련 파일 구조 변경

- visaRegistration 페이지를 src/pages/employee에서 src/pages로 이동
- 관련 기능을 src/features/employee/visaRegistration에서 src/features/visaRegistration으로 이동

* feat: Table 컴포넌트 구현

* feat: 변경된 고용주 마이페이지 구현

* refactor: 회사 관련 공통 기능을 features/companies로 이동 및 CompanyInfo 수정

* refactor: CompanyRecruitments 페이지 이름을 MyCompany로 변경

* feat: EmployerMyAccount path 설정

* refactor: 불필요한 코드 삭제 및 폴더명 일관성 있게 변경

* Feat/#50 Select 컴포넌트 구현 (#53)

* feat: add Select component

* feat: add useGlobalSelect and useSelect

* refactor: move directory

* refactor: EmployerMyAccount 페이지에서 mock 데이터 분리 및 코드 정리

* refactor: visaRegistration 및 applicants 페이지의 mock, style 파일 분리

* refactor: RecruitmentList 컴포넌트 리팩토링 및 RecruitmentsTable 분리

* refactor: CompanyInfo 반응형 디자인 수정

* refactor: 외국인 번호 및 비자 발급 일자 등록 페이지 스타일 수정

* refactor: Header 컴포넌트의 닉네임 버튼을 사용자 프로필 이미지로 변경

* Refactor/#54 Modal 컴포넌트 재설계 (#55)

* chore: add loadable component package

* feat: implement modal management system with context and dynamic loading

* refactor: 코드 리뷰 반영

- formValid를 useMemo로 관리
- validateForeignerNumber 함수를 별도 파일로 분리

* Feat/#56 메인 페이지 API 연동 (#57)

* chore: setting mockServiceWorker

* feat: add useFetchRecruitments hooks and recruitmentsMockHandler

* feat: add useFetchSlides hooks and slidesMockHandler

* feat: add Spinner component

* feat: add AsyncBoundary component

* chore: add msw-storybook-addon

* Feat/#58 OAuth 구글 로그인 구현 (#59)

* feat: add useGoogleOAuth hook

* feat: add Loading page

* chore: add MemoryRouter to decorators

* Feat/#60 가입자 정보 선택 API 연동 (#61)

* feat: add useRegister hook

* fix: change role prop value

* style: Button 컴포넌트 Props 이름변경 theme->design

* feat: 근로자마이페이지 아이콘 설정

* feat: 근로자 마이페이지 구현

* feat: 근로자마이페이지 라우터 설정

* style: Button props 이름 변경

* feat: msw 세팅 및 API path 작성

* feat: 구인글 등록 API 연결 및 msw 세팅

* fix: 구인글 업로드 mock 핸들러 수정

* feat: 근로자 마이페이지 mock 핸들러 추가

* feat: 근로자 마이페이지 API 연결 및 msw 설정

* feat: 이력서 페이지 구현 (#63)

- react-hook-form 을 사용했습니다.
- api 명세서에 맞게 이름,주소,번호,경력,자기소개,한국어실력을 필수값으로 받게 했습니다.



* fix: change button prop

* feat: add GitHub Actions workflow for linting and type checking

* Feat/#74 사인 등록 페이지 구현 (#75)

* 7, 8주차 산출물 (#70)

* Update README.md

* Refactor/#47 6주차 코드리뷰 리팩토링 (#48)

* refactor: separate style and prop-related constants

* refactor: remove lambda function

* refactor: restructure SignUp and RecruitmentHeader components

* feat: separate Button and Text component of SignIn

* feat: add SignUpText component

* refactor: remove auth page's barrel file

* Feat/#42, #43 지원자 목록 페이지 및 팝업 구현 (#51)

* feat: 지원자 목록 페이지 구현

* feat: 지원자 목록 페이지 스토리북 생성

* refactor: ApplicantList 테이블 컴포넌트 분리

* feat: Applicants path 설정

* feat: 지원자 목록 페이지에 계약 관련 팝업 추가

* fix: 이미지 경로 수정 및 불필요한 태그 제거

* refactor: MyAccount 페이지 구조 변경 및 CompanyRecruitments로 파일명 변경

- 기존 MyAccount 페이지를 CompanyRecruitments로 이름 변경
- 새로운 MyAccount 페이지 구현을 위해 기존 페이지의 역할 변경

* refactor: visaRegistration 관련 파일 구조 변경

- visaRegistration 페이지를 src/pages/employee에서 src/pages로 이동
- 관련 기능을 src/features/employee/visaRegistration에서 src/features/visaRegistration으로 이동

* feat: Table 컴포넌트 구현

* feat: 변경된 고용주 마이페이지 구현

* refactor: 회사 관련 공통 기능을 features/companies로 이동 및 CompanyInfo 수정

* refactor: CompanyRecruitments 페이지 이름을 MyCompany로 변경

* feat: EmployerMyAccount path 설정

* refactor: 불필요한 코드 삭제 및 폴더명 일관성 있게 변경

* Feat/#50 Select 컴포넌트 구현 (#53)

* feat: add Select component

* feat: add useGlobalSelect and useSelect

* refactor: move directory

* refactor: EmployerMyAccount 페이지에서 mock 데이터 분리 및 코드 정리

* refactor: visaRegistration 및 applicants 페이지의 mock, style 파일 분리

* refactor: RecruitmentList 컴포넌트 리팩토링 및 RecruitmentsTable 분리

* refactor: CompanyInfo 반응형 디자인 수정

* refactor: 외국인 번호 및 비자 발급 일자 등록 페이지 스타일 수정

* refactor: Header 컴포넌트의 닉네임 버튼을 사용자 프로필 이미지로 변경

* Refactor/#54 Modal 컴포넌트 재설계 (#55)

* chore: add loadable component package

* feat: implement modal management system with context and dynamic loading

* refactor: 코드 리뷰 반영

- formValid를 useMemo로 관리
- validateForeignerNumber 함수를 별도 파일로 분리

* Feat/#56 메인 페이지 API 연동 (#57)

* chore: setting mockServiceWorker

* feat: add useFetchRecruitments hooks and recruitmentsMockHandler

* feat: add useFetchSlides hooks and slidesMockHandler

* feat: add Spinner component

* feat: add AsyncBoundary component

* chore: add msw-storybook-addon

* Feat/#58 OAuth 구글 로그인 구현 (#59)

* feat: add useGoogleOAuth hook

* feat: add Loading page

* chore: add MemoryRouter to decorators

* Feat/#60 가입자 정보 선택 API 연동 (#61)

* feat: add useRegister hook

* fix: change role prop value

* style: Button 컴포넌트 Props 이름변경 theme->design

* feat: 근로자마이페이지 아이콘 설정

* feat: 근로자 마이페이지 구현

* feat: 근로자마이페이지 라우터 설정

* style: Button props 이름 변경

* feat: msw 세팅 및 API path 작성

* feat: 구인글 등록 API 연결 및 msw 세팅

* fix: 구인글 업로드 mock 핸들러 수정

* feat: 근로자 마이페이지 mock 핸들러 추가

* feat: 근로자 마이페이지 API 연결 및 msw 설정

* feat: 이력서 페이지 구현 (#63)

- react-hook-form 을 사용했습니다.
- api 명세서에 맞게 이름,주소,번호,경력,자기소개,한국어실력을 필수값으로 받게 했습니다.



* fix: change button prop

* feat: add GitHub Actions workflow for linting and type checking

---------






* docs: update README.md

* docs: update README.md

* docs: update README.md

* chore: react-signature-canvas 설치

* feat: 사인 등록 페이지 구현
- 사인을 이미지를 파일로 만든 후 서버로 전송합니다.
- 요청 로직도 같이 구현했습니다.
- 요청 완료 후 리다이렉트는 로그인 이후 구현

* feat: 사인 등록 페이지 api 연동(msw)

---------






* Feat/#77 회사 등록 페이지 구현 (#78)

* "refactor: default function 컴포넌트로 변경"

* feat: 회사 등록 페이지 구현

* feat: 회사 등록 페이지 라우터 설정

* style: 회사 이미지 업로드기능 구현

* Feat/#76 지원서 페이지 API 연동(msw) (#79)

* 7, 8주차 산출물 (#70)

* Update README.md

* Refactor/#47 6주차 코드리뷰 리팩토링 (#48)

* refactor: separate style and prop-related constants

* refactor: remove lambda function

* refactor: restructure SignUp and RecruitmentHeader components

* feat: separate Button and Text component of SignIn

* feat: add SignUpText component

* refactor: remove auth page's barrel file

* Feat/#42, #43 지원자 목록 페이지 및 팝업 구현 (#51)

* feat: 지원자 목록 페이지 구현

* feat: 지원자 목록 페이지 스토리북 생성

* refactor: ApplicantList 테이블 컴포넌트 분리

* feat: Applicants path 설정

* feat: 지원자 목록 페이지에 계약 관련 팝업 추가

* fix: 이미지 경로 수정 및 불필요한 태그 제거

* refactor: MyAccount 페이지 구조 변경 및 CompanyRecruitments로 파일명 변경

- 기존 MyAccount 페이지를 CompanyRecruitments로 이름 변경
- 새로운 MyAccount 페이지 구현을 위해 기존 페이지의 역할 변경

* refactor: visaRegistration 관련 파일 구조 변경

- visaRegistration 페이지를 src/pages/employee에서 src/pages로 이동
- 관련 기능을 src/features/employee/visaRegistration에서 src/features/visaRegistration으로 이동

* feat: Table 컴포넌트 구현

* feat: 변경된 고용주 마이페이지 구현

* refactor: 회사 관련 공통 기능을 features/companies로 이동 및 CompanyInfo 수정

* refactor: CompanyRecruitments 페이지 이름을 MyCompany로 변경

* feat: EmployerMyAccount path 설정

* refactor: 불필요한 코드 삭제 및 폴더명 일관성 있게 변경

* Feat/#50 Select 컴포넌트 구현 (#53)

* feat: add Select component

* feat: add useGlobalSelect and useSelect

* refactor: move directory

* refactor: EmployerMyAccount 페이지에서 mock 데이터 분리 및 코드 정리

* refactor: visaRegistration 및 applicants 페이지의 mock, style 파일 분리

* refactor: RecruitmentList 컴포넌트 리팩토링 및 RecruitmentsTable 분리

* refactor: CompanyInfo 반응형 디자인 수정

* refactor: 외국인 번호 및 비자 발급 일자 등록 페이지 스타일 수정

* refactor: Header 컴포넌트의 닉네임 버튼을 사용자 프로필 이미지로 변경

* Refactor/#54 Modal 컴포넌트 재설계 (#55)

* chore: add loadable component package

* feat: implement modal management system with context and dynamic loading

* refactor: 코드 리뷰 반영

- formValid를 useMemo로 관리
- validateForeignerNumber 함수를 별도 파일로 분리

* Feat/#56 메인 페이지 API 연동 (#57)

* chore: setting mockServiceWorker

* feat: add useFetchRecruitments hooks and recruitmentsMockHandler

* feat: add useFetchSlides hooks and slidesMockHandler

* feat: add Spinner component

* feat: add AsyncBoundary component

* chore: add msw-storybook-addon

* Feat/#58 OAuth 구글 로그인 구현 (#59)

* feat: add useGoogleOAuth hook

* feat: add Loading page

* chore: add MemoryRouter to decorators

* Feat/#60 가입자 정보 선택 API 연동 (#61)

* feat: add useRegister hook

* fix: change role prop value

* style: Button 컴포넌트 Props 이름변경 theme->design

* feat: 근로자마이페이지 아이콘 설정

* feat: 근로자 마이페이지 구현

* feat: 근로자마이페이지 라우터 설정

* style: Button props 이름 변경

* feat: msw 세팅 및 API path 작성

* feat: 구인글 등록 API 연결 및 msw 세팅

* fix: 구인글 업로드 mock 핸들러 수정

* feat: 근로자 마이페이지 mock 핸들러 추가

* feat: 근로자 마이페이지 API 연결 및 msw 설정

* feat: 이력서 페이지 구현 (#63)

- react-hook-form 을 사용했습니다.
- api 명세서에 맞게 이름,주소,번호,경력,자기소개,한국어실력을 필수값으로 받게 했습니다.



* fix: change button prop

* feat: add GitHub Actions workflow for linting and type checking

---------






* docs: update README.md

* docs: update README.md

* docs: update README.md

* refactor: 지원서 페이지 버튼 수정
- 버튼 스타일을 수정했습니다.

* feat: 지원서 페이지 api 연동(msw)
- recruitmentId를 Path Parameter로 사용했습니다.
- msw로 성공 요청만 받을 수 있게 했습니다.

---------






* Feat/#69 API 연동 (#80)

* feat: 고용주 마이페이지 API 연결

* feat: 내 회사 페이지 API 연결

* feat: 지원자 목록 페이지 API 연결

* feat: 계약하기 팝업창 API 연결

* feat: 외국인 번호 및 비자 발급 일자 등록 페이지 API 연결

* chore: 의존성 재설치

* chore: 의존성 업데이트

* chore: CI 일관성 확보를 위해 의존성 업데이트

---------



* feat: 근로계약서 path 설정

* Feat/#81 구인 글 상세 화면 수정 및 API 연결 (#83)

* 7, 8주차 산출물 (#70)

* Update README.md

* Refactor/#47 6주차 코드리뷰 리팩토링 (#48)

* refactor: separate style and prop-related constants

* refactor: remove lambda function

* refactor: restructure SignUp and RecruitmentHeader components

* feat: separate Button and Text component of SignIn

* feat: add SignUpText component

* refactor: remove auth page's barrel file

* Feat/#42, #43 지원자 목록 페이지 및 팝업 구현 (#51)

* feat: 지원자 목록 페이지 구현

* feat: 지원자 목록 페이지 스토리북 생성

* refactor: ApplicantList 테이블 컴포넌트 분리

* feat: Applicants path 설정

* feat: 지원자 목록 페이지에 계약 관련 팝업 추가

* fix: 이미지 경로 수정 및 불필요한 태그 제거

* refactor: MyAccount 페이지 구조 변경 및 CompanyRecruitments로 파일명 변경

- 기존 MyAccount 페이지를 CompanyRecruitments로 이름 변경
- 새로운 MyAccount 페이지 구현을 위해 기존 페이지의 역할 변경

* refactor: visaRegistration 관련 파일 구조 변경

- visaRegistration 페이지를 src/pages/employee에서 src/pages로 이동
- 관련 기능을 src/features/employee/visaRegistration에서 src/features/visaRegistration으로 이동

* feat: Table 컴포넌트 구현

* feat: 변경된 고용주 마이페이지 구현

* refactor: 회사 관련 공통 기능을 features/companies로 이동 및 CompanyInfo 수정

* refactor: CompanyRecruitments 페이지 이름을 MyCompany로 변경

* feat: EmployerMyAccount path 설정

* refactor: 불필요한 코드 삭제 및 폴더명 일관성 있게 변경

* Feat/#50 Select 컴포넌트 구현 (#53)

* feat: add Select component

* feat: add useGlobalSelect and useSelect

* refactor: move directory

* refactor: EmployerMyAccount 페이지에서 mock 데이터 분리 및 코드 정리

* refactor: visaRegistration 및 applicants 페이지의 mock, style 파일 분리

* refactor: RecruitmentList 컴포넌트 리팩토링 및 RecruitmentsTable 분리

* refactor: CompanyInfo 반응형 디자인 수정

* refactor: 외국인 번호 및 비자 발급 일자 등록 페이지 스타일 수정

* refactor: Header 컴포넌트의 닉네임 버튼을 사용자 프로필 이미지로 변경

* Refactor/#54 Modal 컴포넌트 재설계 (#55)

* chore: add loadable component package

* feat: implement modal management system with context and dynamic loading

* refactor: 코드 리뷰 반영

- formValid를 useMemo로 관리
- validateForeignerNumber 함수를 별도 파일로 분리

* Feat/#56 메인 페이지 API 연동 (#57)

* chore: setting mockServiceWorker

* feat: add useFetchRecruitments hooks and recruitmentsMockHandler

* feat: add useFetchSlides hooks and slidesMockHandler

* feat: add Spinner component

* feat: add AsyncBoundary component

* chore: add msw-storybook-addon

* Feat/#58 OAuth 구글 로그인 구현 (#59)

* feat: add useGoogleOAuth hook

* feat: add Loading page

* chore: add MemoryRouter to decorators

* Feat/#60 가입자 정보 선택 API 연동 (#61)

* feat: add useRegister hook

* fix: change role prop value

* style: Button 컴포넌트 Props 이름변경 theme->design

* feat: 근로자마이페이지 아이콘 설정

* feat: 근로자 마이페이지 구현

* feat: 근로자마이페이지 라우터 설정

* style: Button props 이름 변경

* feat: msw 세팅 및 API path 작성

* feat: 구인글 등록 API 연결 및 msw 세팅

* fix: 구인글 업로드 mock 핸들러 수정

* feat: 근로자 마이페이지 mock 핸들러 추가

* feat: 근로자 마이페이지 API 연결 및 msw 설정

* feat: 이력서 페이지 구현 (#63)

- react-hook-form 을 사용했습니다.
- api 명세서에 맞게 이름,주소,번호,경력,자기소개,한국어실력을 필수값으로 받게 했습니다.



* fix: change button prop

* feat: add GitHub Actions workflow for linting and type checking

---------






* docs: update README.md

* docs: update README.md

* docs: update README.md

* refactor: 구인글 상세 정보 반환 페이지 수정
- 데이터 타입을 api 명세에 맞게 변경했습니다.
- recruitmentsDetail 컴포넌트의 스타일을 변경했습니다.
- 타입을 별도로 분리했습니다.

* feat: 구인글 상세화면 api 연동(msw)
- api 명세서에 맞게 요청 로직을 구현했습니다.
- msw를 통해 반환 값을 받을 수 있게 했습니다.

---------






* Refactor/#71 9주차 기능 개선 및 리팩토링 (#84)

* refactor: 컴포넌트 이름과 일치하도록 파일 이름 변경

* refactor: 고용주 마이페이지에 사인등록 버튼 추가 및 디자인 변경

* feat: 고용주 마이페이지의 사인 등록 버튼 클릭 시 사인 등록 페이지로 이동 기능 추가

* feat: 회사 목록 로딩 중 Spinner 추가 및 위치 조정

* feat: 공고글 목록 로딩 중 Spinner 추가

* refactor: ContractModal의 버튼들을 ModalButtons로 분리

* feat: 계약 팝업창 확인 버튼 클릭 시 근로계약서 페이지로 이동 기능 추가

* Feat/#85 Header 컴포넌트 기능 추가 및 개선 (#87)

* feat: 로그인 여부 및 사용자 타입에 따라 Header 구성 변경

- 로그인 여부에 따라 로그인/로그아웃 버튼 표시
- 고용주인 경우 '채용공고 등록' 버튼 추가
- 로컬스토리지에서 사용자 정보를 가져와 반영

* feat: Header 네비게이션 기능 추가

- 로고 클릭 시 홈 화면으로 이동
- 로그인 버튼 클릭 시 로그인 페이지로 이동
- 채용공고 등록 버튼 클릭 시 구인글 업로드 페이지로 이동
- 프로필 이미지 클릭 시 마이페이지로 이동

* feat: 근로자 마이페이지 path 연결 완료

* feat: 구인글 업로드 path 연결 완료

* feat: 회사 등록 API 연결

* feat: i18n 번역 데이터 - 구인글 업로드, 근로자 마이페이지, 회사 등록 페이지

* refactor: 구인글 업로드 및 근로자 마이페이지 폴더구조 정리

* feat: 고용주 근로계약서 작성 API 연결 및 구인글 작성 API 폴더구조 정리

* feat: 근로자 근로계약서 불러오기 API 연결

* feat: 근로자 근로계약서 저장하기 API 연결

---------

Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>
Co-authored-by: LEE YONGJIN <129606040+tteokbokki-master@users.noreply.github.com>

JunilHwang commented on package.json:
```diff
@@ -31,13 +31,17 @@
     "@loadable/component": "^5.16.4",
     "@react-oauth/google": "^0.12.1",
     "@tanstack/react-query": "^5.56.2",
+    "@types/react-signature-canvas": "^1.0.6",
```
types 패키지가 dependencies 에 설치되어있네요..!

JunilHwang commented on package.json:
```diff
@@ -31,13 +31,17 @@
     "axios": "^1.7.7",
+    "buffer": "^6.0.3",
     "csstype": "^3.1.3",
+    "jquery": "^3.7.1",
```
jquery가 필요한 상황이 맞을까요?

JunilHwang commented on src/types/index.d.ts:
```diff
@@ -23,18 +23,28 @@ export type ForeignerData = {
 
-export type StateProps = '근로계약서 서명하기' | '채용 마감' | '지원서 검토중' | '채용 완료';
+// 백엔드에서 정하는 값에 따라 key 바꾸면 됨
+export const State = {
+  LetsSign: '근로계약서 서명하기',
+  Closed: '채용 마감',
+  Waiting: '지원서 검토중',
+  Completed: '채용 완료',
+} as const;
```
`index.d.ts` 에는 타입만 정의가 되어있어야 하는데, 이렇게 상수가 정의되어있네요..!

JunilHwang commented on README.md:
```diff
@@ -1,60 +1,26 @@
-이처럼, 여러 값만 도메인으로 설정한 타입들에 한해 자주 에러가 발생하는데 이런 부분을 어떻게 해결하면 좋을까요?
-
-- 고용주 마이페이지, 내 회사 페이지, 지원자 목록 페이지에 공통으로 사용되는 테이블 컴포넌트를 구현했습니다. 세 페이지 모두 같은 스타일의 테이블을 사용하여 구현 시 스타일을 컴포넌트 자체에 내장하여 일괄적으로 적용되도록 했는데, 이런 방식으로 사용해도 될지 궁금합니다. 스타일을 고정하여 일관성을 유지하는 것과 사용 시점에 스타일을 적용해서 유연성을 제공하는 방법 중에 어느 것을 선택하는 것이 좋을까요?
+        <Section>
+          <Typo bold element="h3" size="20px" style={{ marginBottom: '24px' }}>
+            내가 지원한 공고
+          </Typo>
+          {myRecruitList && <MyRecruitList myRecruitList={myRecruitList} />}
+        </Section>
```
두 가지 방법을 고민해볼 수 있을 것 같아요.

1. 특별한 경우가 아니라면, 일단 모든 컴포넌트는 다 분리한다.
2. style 관련 컴포넌트는 `Styles.tsx` 처럼 일관된 이름으로 분리한다.

아니면 FSD 아키텍쳐를 사용하는 방법도 있을 것 같네요!

일단 지금 당장은 현재 방식처럼 해도 큰 문제는 없어보입니다.
오히려 Section 같은게 다른 파일로 분리되어있으면 컨텍스트를 파악하는게 더 헷갈릴 수 있어요.

JunilHwang commented on src/apis/applicants/mocks/foreignerMockHandler.ts:
```diff
@@ -0,0 +1,5 @@
+import { APIPath } from '@/apis/apiPath';
+import { foreigner } from '@/features/applicants/ApplicantList/ContractModal/ContractModal.mock';
+import { http, HttpResponse } from 'msw';
+
+export const foreignerMockHandler = [http.get(APIPath.getForeigner, () => HttpResponse.json(foreigner))];
```
API 영역에서 features를 참조하고 있는건 이상해보여요. 
features에서 api를 참고하도록 만들어야 하지 않을까요?

아니면 이 mocks 폴더 자체를 분리해야 하지 않을까요?

JunilHwang commented on src/apis/applicants/mocks/myApplicantsMockHandler.ts:
```diff
@@ -0,0 +1,5 @@
+import { APIPath } from '@/apis/apiPath';
+import { applicantList } from '@/pages/applicants/Applicants.mock';
+import { http, HttpResponse } from 'msw';
+
+export const myApplicantsMockHandler = [http.get(APIPath.getMyApplicants, () => HttpResponse.json(applicantList))];
```
여긴 심지어 pages를 참고하는군요..

JunilHwang commented on src/apis/apply/useApplyHook.tsx:
```diff
@@ -0,0 +1,23 @@
+import { APIPath } from '@/apis/apiPath';
+import { clientInstance } from '@/apis/instance';
```
지금 apis에서 apis를 참고하는 형태로 작성이 되어있답니다.
같은 폴더(apis) 내에서는 상대경로를 사용할 수 있도록 해주세요..!

JunilHwang commented on src/apis/companies/mocks/myCompaniesMockHandler.ts:
```diff
@@ -0,0 +1,5 @@
+import { APIPath } from '@/apis/apiPath';
+import { companyList } from '@/pages/myPage/employer/EmployerMyPage.mock';
```
앞선 리뷰와 동일합니다!

JunilHwang commented on src/apis/postNotice/hooks/usePostNotice.ts:
```diff
@@ -0,0 +1,32 @@
+import { clientInstance } from '@/apis/instance';
+import { useMutation } from '@tanstack/react-query';
+
+export type NoticeRequestData = {
```
내부에서만 쓰이고 있어서 export를 해줄 필요는 없어보여요..!

JunilHwang commented on src/apis/postNotice/hooks/usePostNotice.ts:
```diff
@@ -0,0 +1,32 @@
+export const useFetchPostNotice = () =>
+  useMutation({
+    mutationFn: postNotice,
+  });
```
fetchPost 라고 표현하는게 맞을까요..!?

JunilHwang commented on src/apis/postNotice/hooks/usePostNotice.ts:
```diff
@@ -0,0 +1,32 @@
+  return response.data;
+};
+
+export const useFetchPostNotice = () =>
```
이제보니 파일 이름은 usePostNotice인데, 실제로 선언된 함수는 useFetchPostNotice 로 되어있네요..
이런 부분들도 일치할 수 있도록 신경써주세요!

JunilHwang commented on src/apis/recruitmentsDetail/recruitmentsDetailMockHandler.ts:
```diff
@@ -0,0 +1,7 @@
+import { recruitmentDetailList } from './recruitmentsDetailList';
+import { http, HttpResponse } from 'msw';
+
+export const recruitmentsDetailMockHandler = [
+  http.get(APIPath.recruitmentsDetail, () => HttpResponse.json(recruitmentDetailList)),
+];
```
전체적으로 목킹을 위한 파일과 어플리케이션을 위한 파일이 너무 섞여있는 것 같아요..!
msw를 사용하는 것은 좋지만
msw에 대한 부분을 잘 분리해서 관리할 수 있도록 해주면 좋을 것 같습니다 ㅎㅎ

JunilHwang commented on src/apis/recruitmentsDetail/useRecruitmentsDetail.tsx:
```diff
@@ -0,0 +1,15 @@
+import { getDynamicAPIPath } from '@/apis/apiPath';
+import { clientInstance } from '@/apis/instance';
+import { useQuery } from '@tanstack/react-query';
+import { type RecruitProps } from '@/pages/recruit/RecruitType';
```
마찬가지로.. pages를 의존하지 않도록 해주세요.

JunilHwang commented on src/apis/registerCompany/useRegisterCompany.ts:
```diff
@@ -0,0 +1,23 @@
+  return response.data;
+};
+
+export const useFetchPostCompany = () =>
+  useMutation({
+    mutationFn: postCompany,
+  });
```
여기도 파일 이름이랑 훅 이름이 일치하지가 않네요..

JunilHwang commented on src/apis/registerSign/useRegisterSign.ts:
```diff
@@ -0,0 +1,27 @@
+  return response.data;
+};
+
+export const FetchRegisterSign = () => {
+  // const nav = useNavigate();
+
+  return useMutation({
+    mutationFn: (req: FormData) => RegisterSign(req),
+    onSuccess: () => {
+      // 고용주, 근로자 구분 리다이렉트 추후 구현
+      // nav('/');
+    },
+  });
+};
```
여기는 아예 use가 안붙어 있군요 ㅎㅎ;

JunilHwang commented on src/assets/translator/EmployeeMyPage/ko.ts:
```diff
@@ -0,0 +1,7 @@
+export const EmployeeMyPage = {
+  UPDATE_PROFILE: '프로필 수정하기',
+  REGISTER_RESUME: '이력서 등록하기',
+  REGISTER_SIGN: '사인 등록',
+  REGISTER_VISA: '외국인 번호 및 비자 발급 일자 등록',
+  MYRECRUITLIST: '내가 지원한 공고',
+};
```
똑같은 변수의 이름이 ko에도 있고 ve에도 있어요.
이런식으로 표현해주면 어떨까요?

```tsx
const Languages = {
    KO: 'ko',
    VE: 've',
} as const ;

const EmployeeMyPage = {
  [Languages.KO]: { /* 생략 */ },
  [Languages.VE]: { /* 생략 */ },
}
```

JunilHwang commented on src/features/applicants/ApplicantList/ApplicantsTable/ApplicantsTable.tsx:
```diff
@@ -0,0 +1,61 @@
+          />
+        </tbody>
+      </Table>
+      {selectedUserId && <ContractModal isOpen={isModalOpen} onClose={handleCloseModal} userId={selectedUserId} />}
```
항상 반복적으로 하는 이야기이긴 한데..
onClose 의 동작이 절대 아닙니다..

JunilHwang commented on src/features/applicants/ApplicantList/ApplicantsTable/ApplicantsTable.tsx:
```diff
@@ -0,0 +1,61 @@
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [selectedUserId, setSelectedUserId] = useState<number | null>(null);
+
+  const handleOpenModal = (userId: number) => {
+    setSelectedUserId(userId);
+    setIsModalOpen(true);
+  };
+
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
+    setSelectedUserId(null);
+  };
```
모달 밖에서 close, open 을 해주는데 어떻게 내부에서 onClose 이벤트가 발생할 수 있는걸까요..

JunilHwang commented on src/features/applicants/ApplicantList/ApplicantsTable/ApplicantsTable.tsx:
```diff
@@ -0,0 +1,61 @@
+                  <Flex justifyContent="flex-end" alignItems="center" gap={{ x: '20px' }} css={buttonGroupStyle}>
+                    <Button css={buttonStyle}>이력서</Button>
+                    <Button css={buttonStyle}>지원서</Button>
+                    <Button css={buttonStyle} onClick={() => handleOpenModal(applicant.userId)}>
```
handle 이라는 prefix를 남발하지 않도록 해주세요 ㅎㅎ
handleOpenModal이라는 이름이 이 모습에는 적합하지 않아요
그냥 openModal 로 표현해야 자연스러워보입니다.

JunilHwang commented on src/features/applicants/ApplicantList/ContractModal/ModalButtons/ModalButtons.tsx:
```diff
@@ -0,0 +1,30 @@
+export default function ModalButtons({ onClose }: ModalButtonsProps) {
+  const navigate = useNavigate();
+
+  const handleConfirmClick = () => {
+    navigate(ROUTE_PATH.CONTRACT.EMPLOYER);
+  };
```
이 경우에도
handleConfirmClick
 처럼 표현할 수도 있지만,
`goToEmployer` 처럼 표현할 수도 있겠죠?

JunilHwang commented on src/features/applicants/ApplicantList/ContractModal/ModalButtons/ModalButtons.tsx:
```diff
@@ -0,0 +1,30 @@
+import { Button, Flex, Icon, Typo } from '@/components/common';
+import { buttonTextStyle, customButtonStyle } from './ModalButtons.styles';
+import { useNavigate } from 'react-router-dom';
+import ROUTE_PATH from '@/routes/path';
```
feature에서 routes를 참고하는게 자연스러운 모습일까요?

JunilHwang commented on src/features/applicants/ApplicantList/ContractModal/ContractModal.tsx:
```diff
@@ -0,0 +1,29 @@
+import { Modal } from '@/components/common';
+import ModalText from './ModalText/ModalText';
```
`ModalText/ModalText` 처럼 경로가 표현되는건 자연스럽지 못한 것 같아요..!

JunilHwang commented on src/features/applicants/ApplicantList/ContractModal/ContractModal.tsx:
```diff
@@ -0,0 +1,29 @@
+import { Modal } from '@/components/common';
+import ModalText from './ModalText/ModalText';
+import { useGetForeigner } from '@/apis/applicants/hooks/useGetForeigner';
+import ModalButtons from './ModalButtons/ModalButtons';
```
이 경우도 동일합니다.

JunilHwang commented on src/features/companies/CompanyList/CompaniesTable/CompaniesTable.tsx:
```diff
@@ -0,0 +1,53 @@
+export default function CompaniesTable({ companyList }: Props) {
+  const navigate = useNavigate();
+
+  const handleCompanyClick = (companyId: number) => {
+    navigate(ROUTE_PATH.MY_COMPANY.replace(':companyId', companyId.toString()));
+  };
```
여기도 꼭 handle 이 붙어야 하는지 고민해보시면 좋겠어요.
`goToCompanyPage` 처럼 표현하면 어떨지..

JunilHwang commented on src/features/layout/Header/components/LanguageFilter.tsx:
```diff
@@ -1,8 +1,20 @@
 import theme from '@/assets/theme';
 import { Select, Icon, List } from '@/components/common';
 import useGlobalSelect from '@/components/common/Select/hooks/useGlobalSelect';
```
```tsx
import { Select, Icon, List, useGlobalSelect } from '@/components/common';
```
이런식으로 가져오도록 해주면 이상할까요?

JunilHwang commented on src/features/layout/Header/index.tsx:
```diff
@@ -15,28 +15,80 @@ import {
   const [menuOpen, toggleMenu] = useToggle();
+  const [user, setUser] = useState<UserData | null>(null);
+
+  useEffect(() => {
+    const userData = localStorage.getItem('user');
+
+    if (userData) {
+      setUser(JSON.parse(userData));
+    }
+  }, []);
```
localStorage를 직접 접근하고 있네요. 추상화해서 사용하면 좋겠어요.

JunilHwang commented on src/features/layout/Header/index.tsx:
```diff
@@ -15,28 +15,80 @@ import {
+    }
+  }, []);
+
+  const handleLogoClick = () => {
+    startTransition(() => {
+      navigate(ROUTE_PATH.HOME);
+    });
+  };
+
+  const handleLogin = () => {
+    navigate(ROUTE_PATH.AUTH.SIGN_IN);
+  };
+
+  const handleLogout = () => {
+    localStorage.removeItem('token');
+    localStorage.removeItem('user');
+    setUser(null);
+    navigate(ROUTE_PATH.HOME);
+  };
+
+  const handlePostNotice = () => {
+    navigate(ROUTE_PATH.POST_NOTICE);
+  };
+
+  const handleProfileClick = () => {
+    if (user?.type === 'employer') {
+      navigate(ROUTE_PATH.MY_PAGE.EMPLOYER);
+    } else if (user?.type === 'employee') {
+      navigate(ROUTE_PATH.EMPLOYEE.EMPLOYEE_PAGE);
+    }
+  };
```
여기에 선언된 모든 것들은 전부 이벤트를 다루고 있지 않아요.
handle로 쓰이는게 맞을까요?

JunilHwang commented on src/features/recruitments/RecruitmentInfo/RecruitmentInfo.tsx:
```diff
@@ -0,0 +1,50 @@
+  recruitmentFlexStyle,
+} from './RecruitmentInfo.styles';
+
+type Props = Pick<RecruitmentItem, 'image' | 'companyName' | 'koreanTitle' | 'area' | 'salary'>;
```
차라리 Pick 대신에 Omit을 해주시면...

JunilHwang commented on src/features/layout/Header/index.tsx:
```diff
@@ -15,28 +15,80 @@ import {
-type Props = Pick<UserData, 'profileImage'>;
-
-export default function Header({ profileImage }: Props) {
+export default function Header() {
```
이제보니 Layout이 Feature에 있네요... ㅎㅎ

JunilHwang replied:
어딘가에서는 index.tsx에 컴포넌트가 선언되어있군요..

차라리

```tsx
// index.ts
export { default as Header } from './Header'
```

이렇게 index파일을 하나 선언해주고 컴포넌트 파일을 따로 선언해주면 관리하기 더 좋지 않을까요?

JunilHwang commented on src/features/registerVisa/validateForeignerNumber.ts:
```diff
@@ -0,0 +1,4 @@
+export const validateForeignerNumber = (number: string) => {
+  const regex = /^\d{6}-\d{7}$/;
+  return regex.test(number);
+};
```
이런 친구들에 대해선 간단하게라도 테스트가 작성되어있으면 좋겠어요.

JunilHwang commented on src/features/registerVisa/VisaRegistrationForm.tsx:
```diff
@@ -0,0 +1,95 @@
+import { Button, Flex, Input, Modal } from '@/components/common';
+import { ChangeEvent, useMemo, useState } from 'react';
+<<<<<<<< HEAD:src/features/registerVisa/VisaRegistrationForm.tsx
+import { buttonStyle, ErrorMessage, Form, inputStyle } from './VisaRegistrationForm.styles';
+import { validateForeignerNumber } from './validateForeignerNumber';
+import { useRegisterVisaInfo } from '@/apis/applicants/hooks/useRegisterVisaInfo';
+========
+import { buttonStyle, ErrorMessage, Form, inputStyle } from './index.styles';
+import { validateForeignerNumber } from './validateForeignerNumber';
+>>>>>>>> Develop:src/features/visaRegistration/index.tsx
+
+export default function VisaRegistrationForm() {
+  const [foreignerIdNumber, setForeignerNumber] = useState('');
+  const [visaGenerateDate, setVisaGenerateDate] = useState('');
+  const [error, setError] = useState('');
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const formValid = useMemo(() => !error, [error]);
+<<<<<<<< HEAD:src/features/registerVisa/VisaRegistrationForm.tsx
+
+  const registerVisaMutation = useRegisterVisaInfo();
+========
+>>>>>>>> Develop:src/features/visaRegistration/index.tsx
```
충돌이 제대로 해결되지 못했네요...

JunilHwang commented on src/features/visaRegistration/index.tsx:
```diff
@@ -1,14 +1,25 @@
 import { Button, Flex, Input, Modal } from '@/components/common';
 import { ChangeEvent, useMemo, useState } from 'react';
+<<<<<<<< HEAD:src/features/registerVisa/VisaRegistrationForm.tsx
+import { buttonStyle, ErrorMessage, Form, inputStyle } from './VisaRegistrationForm.styles';
+import { validateForeignerNumber } from './validateForeignerNumber';
+import { useRegisterVisaInfo } from '@/apis/applicants/hooks/useRegisterVisaInfo';
+========
 import { buttonStyle, ErrorMessage, Form, inputStyle } from './index.styles';
 import { validateForeignerNumber } from './validateForeignerNumber';
+>>>>>>>> Develop:src/features/visaRegistration/index.tsx
```
여기도..

JunilHwang commented on src/features/visaRegistration/index.tsx:
```diff
@@ -22,7 +33,19 @@ export default function VisaRegistrationForm() {
+      { foreignerIdNumber, visaGenerateDate },
+      {
+        onSuccess: () => {
+          setForeignerNumber('');
+          setVisaGenerateDate('');
+          setIsModalOpen(true);
```
동일한 라이프 사이클을 가진 state의 경우 하나로 묶어주거나 useReducer로 묶어주면 어떨까요?

JunilHwang commented on src/pages/myCompany/MyCompany.tsx:
```diff
@@ -0,0 +1,71 @@
+  recruitmentList?: RecruitmentItem[];
+}
+
+export default function MyCompany({ company, recruitmentList }: MyCompanyProps) {
+  const { companyId } = useParams();
+  const { data: companyList } = useGetMyCompanies();
+  const { data: recruitments, isLoading } = useGetMyRecruitments(Number(companyId));
+
+  const companyData = company || companyList?.find((c: CompanyData) => c.companyId.toString() === companyId);
+  const recruitmentsData = recruitmentList || recruitments;
```
company랑 recruitmentList가 왜 인자로 있을까 봤더니.. 스토리북에서 쓰이기 위함이군요 ㅎㅎ
지금 당장은 괜찮아보이지만 적절한 방법은 아닌 것 같아요..!
차라리 MyCompanyPage와 MyCompany를 분리해서 관리해주면 어땠을까 싶네요.

MyCompanyPage는 직접 api를 호출하는 방식으로 구성된 컴포넌트고
데이터를 MyCompany에게 넘겨주도록 하고
MyCompany는 props를 통해서만 보여주도록.

JunilHwang commented on src/pages/recruit/RecruitType.ts:
```diff
@@ -0,0 +1,38 @@
+  companyName: string;
+  koreanDetailedDescription: string;
+  vietnameseDetailedDescription: string;
+}
```
Item이 현재 파일에서만 쓰이는데 export를 해줄 필요가 있을까요?

JunilHwang commented on src/pages/registerCompany/RegisterCompany.tsx:
```diff
@@ -0,0 +1,114 @@
+  const mutation = useFetchPostCompany();
+  const navigate = useNavigate();
+
+  const [inputs, setInputs] = useState({
+    name: '',
+    industryOccupation: '',
+    brand: '',
+    revenuePerYear: 0,
+    logoImage: '',
+  });
```
컴포넌트 밖에 inputs에 대한 기본값을 하나 정의해주면 좋겠어요..!

```tsx
const [inputs, setInputs] = useState({ ...DEFAULT_INPUTS });
```

이러면 나중에 초기화할 때도 편하답니다!

JunilHwang commented on src/pages/registerSign/RegisterSign.tsx:
```diff
@@ -0,0 +1,91 @@
+export default function RegisterSign() {
+  const signCanvas = useRef<ReactSignatureCanvas>(null);
+
+  const clear = () => {
+    signCanvas.current?.clear();
+  };
+  const mutation = FetchRegisterSign();
+
+  const convertDataUrlToFile = () => {
+    const dataURL = signCanvas.current?.toDataURL('image/png');
+    const decodedURL = dataURL?.replace(/^data:image\/\w+;base64,/, '') as string;
+    const buf = Buffer.from(decodedURL, 'base64');
+    const blob = new Blob([buf], { type: 'image/png' });
+    return new File([blob], `signature_${Date.now()}.png`, { type: 'image/png' });
+  };
+
+  const onSave = () => {
+    const file = convertDataUrlToFile();
+    const formData = new FormData();
+    formData.append('imageUrl', file);
+    mutation.mutate(formData);
+  };
```
onSave -> save

JunilHwang replied:
현재 컴포넌트가 state를 의존하는 구간이 없다보니  clear, convertDataUrlToFile, onSave 등에는 useCallback 같은게 씌워져있으면 좋겠어요.

JunilHwang reviewed (approved):
전체적으로 기능은 잘 만들어져가는 중인 것 같아요!
피드백을 종합해보면

1. 이벤트 핸들러와 함수를 구분해주기.
2. 파일 이름과 내부 함수 이름을 일치시켜주기.
3. 각 폴더간의 의존성을 신경쓰기
4. 목킹 잘 분리하기
5. 불필요한 export 제거하기

이정도인 것 같네요..!
어플리케이션에 대한 테스트는 지금 불가능해서... 이에 대한 피드백을 드리기가 힘든 상황입니다 ㅠㅠ (충돌 해결이 안 되어있음 + 스토리북에는 인터랙션이 제대로 동작하지 않음)

이번주도 고생하셨어요!

JunilHwang reviewed (changes requested):
충돌 해결 후에 머지해주세요~

kang-kibong committed:
Develop branch merge to Master branch (#98)

* 9주차 산출물  (#91)

* Update README.md

* Refactor/#47 6주차 코드리뷰 리팩토링 (#48)

* refactor: separate style and prop-related constants

* refactor: remove lambda function

* refactor: restructure SignUp and RecruitmentHeader components

* feat: separate Button and Text component of SignIn

* feat: add SignUpText component

* refactor: remove auth page's barrel file

* Feat/#42, #43 지원자 목록 페이지 및 팝업 구현 (#51)

* feat: 지원자 목록 페이지 구현

* feat: 지원자 목록 페이지 스토리북 생성

* refactor: ApplicantList 테이블 컴포넌트 분리

* feat: Applicants path 설정

* feat: 지원자 목록 페이지에 계약 관련 팝업 추가

* fix: 이미지 경로 수정 및 불필요한 태그 제거

* refactor: MyAccount 페이지 구조 변경 및 CompanyRecruitments로 파일명 변경

- 기존 MyAccount 페이지를 CompanyRecruitments로 이름 변경
- 새로운 MyAccount 페이지 구현을 위해 기존 페이지의 역할 변경

* refactor: visaRegistration 관련 파일 구조 변경

- visaRegistration 페이지를 src/pages/employee에서 src/pages로 이동
- 관련 기능을 src/features/employee/visaRegistration에서 src/features/visaRegistration으로 이동

* feat: Table 컴포넌트 구현

* feat: 변경된 고용주 마이페이지 구현

* refactor: 회사 관련 공통 기능을 features/companies로 이동 및 CompanyInfo 수정

* refactor: CompanyRecruitments 페이지 이름을 MyCompany로 변경

* feat: EmployerMyAccount path 설정

* refactor: 불필요한 코드 삭제 및 폴더명 일관성 있게 변경

* Feat/#50 Select 컴포넌트 구현 (#53)

* feat: add Select component

* feat: add useGlobalSelect and useSelect

* refactor: move directory

* refactor: EmployerMyAccount 페이지에서 mock 데이터 분리 및 코드 정리

* refactor: visaRegistration 및 applicants 페이지의 mock, style 파일 분리

* refactor: RecruitmentList 컴포넌트 리팩토링 및 RecruitmentsTable 분리

* refactor: CompanyInfo 반응형 디자인 수정

* refactor: 외국인 번호 및 비자 발급 일자 등록 페이지 스타일 수정

* refactor: Header 컴포넌트의 닉네임 버튼을 사용자 프로필 이미지로 변경

* Refactor/#54 Modal 컴포넌트 재설계 (#55)

* chore: add loadable component package

* feat: implement modal management system with context and dynamic loading

* refactor: 코드 리뷰 반영

- formValid를 useMemo로 관리
- validateForeignerNumber 함수를 별도 파일로 분리

* Feat/#56 메인 페이지 API 연동 (#57)

* chore: setting mockServiceWorker

* feat: add useFetchRecruitments hooks and recruitmentsMockHandler

* feat: add useFetchSlides hooks and slidesMockHandler

* feat: add Spinner component

* feat: add AsyncBoundary component

* chore: add msw-storybook-addon

* Feat/#58 OAuth 구글 로그인 구현 (#59)

* feat: add useGoogleOAuth hook

* feat: add Loading page

* chore: add MemoryRouter to decorators

* Feat/#60 가입자 정보 선택 API 연동 (#61)

* feat: add useRegister hook

* fix: change role prop value

* style: Button 컴포넌트 Props 이름변경 theme->design

* feat: 근로자마이페이지 아이콘 설정

* feat: 근로자 마이페이지 구현

* feat: 근로자마이페이지 라우터 설정

* style: Button props 이름 변경

* feat: msw 세팅 및 API path 작성

* feat: 구인글 등록 API 연결 및 msw 세팅

* fix: 구인글 업로드 mock 핸들러 수정

* feat: 근로자 마이페이지 mock 핸들러 추가

* feat: 근로자 마이페이지 API 연결 및 msw 설정

* feat: 이력서 페이지 구현 (#63)

- react-hook-form 을 사용했습니다.
- api 명세서에 맞게 이름,주소,번호,경력,자기소개,한국어실력을 필수값으로 받게 했습니다.

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>

* fix: change button prop

* feat: add GitHub Actions workflow for linting and type checking

* Feat/#74 사인 등록 페이지 구현 (#75)

* 7, 8주차 산출물 (#70)

* Update README.md

* Refactor/#47 6주차 코드리뷰 리팩토링 (#48)

* refactor: separate style and prop-related constants

* refactor: remove lambda function

* refactor: restructure SignUp and RecruitmentHeader components

* feat: separate Button and Text component of SignIn

* feat: add SignUpText component

* refactor: remove auth page's barrel file

* Feat/#42, #43 지원자 목록 페이지 및 팝업 구현 (#51)

* feat: 지원자 목록 페이지 구현

* feat: 지원자 목록 페이지 스토리북 생성

* refactor: ApplicantList 테이블 컴포넌트 분리

* feat: Applicants path 설정

* feat: 지원자 목록 페이지에 계약 관련 팝업 추가

* fix: 이미지 경로 수정 및 불필요한 태그 제거

* refactor: MyAccount 페이지 구조 변경 및 CompanyRecruitments로 파일명 변경

- 기존 MyAccount 페이지를 CompanyRecruitments로 이름 변경
- 새로운 MyAccount 페이지 구현을 위해 기존 페이지의 역할 변경

* refactor: visaRegistration 관련 파일 구조 변경

- visaRegistration 페이지를 src/pages/employee에서 src/pages로 이동
- 관련 기능을 src/features/employee/visaRegistration에서 src/features/visaRegistration으로 이동

* feat: Table 컴포넌트 구현

* feat: 변경된 고용주 마이페이지 구현

* refactor: 회사 관련 공통 기능을 features/companies로 이동 및 CompanyInfo 수정

* refactor: CompanyRecruitments 페이지 이름을 MyCompany로 변경

* feat: EmployerMyAccount path 설정

* refactor: 불필요한 코드 삭제 및 폴더명 일관성 있게 변경

* Feat/#50 Select 컴포넌트 구현 (#53)

* feat: add Select component

* feat: add useGlobalSelect and useSelect

* refactor: move directory

* refactor: EmployerMyAccount 페이지에서 mock 데이터 분리 및 코드 정리

* refactor: visaRegistration 및 applicants 페이지의 mock, style 파일 분리

* refactor: RecruitmentList 컴포넌트 리팩토링 및 RecruitmentsTable 분리

* refactor: CompanyInfo 반응형 디자인 수정

* refactor: 외국인 번호 및 비자 발급 일자 등록 페이지 스타일 수정

* refactor: Header 컴포넌트의 닉네임 버튼을 사용자 프로필 이미지로 변경

* Refactor/#54 Modal 컴포넌트 재설계 (#55)

* chore: add loadable component package

* feat: implement modal management system with context and dynamic loading

* refactor: 코드 리뷰 반영

- formValid를 useMemo로 관리
- validateForeignerNumber 함수를 별도 파일로 분리

* Feat/#56 메인 페이지 API 연동 (#57)

* chore: setting mockServiceWorker

* feat: add useFetchRecruitments hooks and recruitmentsMockHandler

* feat: add useFetchSlides hooks and slidesMockHandler

* feat: add Spinner component

* feat: add AsyncBoundary component

* chore: add msw-storybook-addon

* Feat/#58 OAuth 구글 로그인 구현 (#59)

* feat: add useGoogleOAuth hook

* feat: add Loading page

* chore: add MemoryRouter to decorators

* Feat/#60 가입자 정보 선택 API 연동 (#61)

* feat: add useRegister hook

* fix: change role prop value

* style: Button 컴포넌트 Props 이름변경 theme->design

* feat: 근로자마이페이지 아이콘 설정

* feat: 근로자 마이페이지 구현

* feat: 근로자마이페이지 라우터 설정

* style: Button props 이름 변경

* feat: msw 세팅 및 API path 작성

* feat: 구인글 등록 API 연결 및 msw 세팅

* fix: 구인글 업로드 mock 핸들러 수정

* feat: 근로자 마이페이지 mock 핸들러 추가

* feat: 근로자 마이페이지 API 연결 및 msw 설정

* feat: 이력서 페이지 구현 (#63)

- react-hook-form 을 사용했습니다.
- api 명세서에 맞게 이름,주소,번호,경력,자기소개,한국어실력을 필수값으로 받게 했습니다.

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>

* fix: change button prop

* feat: add GitHub Actions workflow for linting and type checking

---------

Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>
Co-authored-by: LEE YONGJIN <129606040+tteokbokki-master@users.noreply.github.com>

* docs: update README.md

* docs: update README.md

* docs: update README.md

* chore: react-signature-canvas 설치

* feat: 사인 등록 페이지 구현
- 사인을 이미지를 파일로 만든 후 서버로 전송합니다.
- 요청 로직도 같이 구현했습니다.
- 요청 완료 후 리다이렉트는 로그인 이후 구현

* feat: 사인 등록 페이지 api 연동(msw)

---------

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>
Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>

* Feat/#77 회사 등록 페이지 구현 (#78)

* "refactor: default function 컴포넌트로 변경"

* feat: 회사 등록 페이지 구현

* feat: 회사 등록 페이지 라우터 설정

* style: 회사 이미지 업로드기능 구현

* Feat/#76 지원서 페이지 API 연동(msw) (#79)

* 7, 8주차 산출물 (#70)

* Update README.md

* Refactor/#47 6주차 코드리뷰 리팩토링 (#48)

* refactor: separate style and prop-related constants

* refactor: remove lambda function

* refactor: restructure SignUp and RecruitmentHeader components

* feat: separate Button and Text component of SignIn

* feat: add SignUpText component

* refactor: remove auth page's barrel file

* Feat/#42, #43 지원자 목록 페이지 및 팝업 구현 (#51)

* feat: 지원자 목록 페이지 구현

* feat: 지원자 목록 페이지 스토리북 생성

* refactor: ApplicantList 테이블 컴포넌트 분리

* feat: Applicants path 설정

* feat: 지원자 목록 페이지에 계약 관련 팝업 추가

* fix: 이미지 경로 수정 및 불필요한 태그 제거

* refactor: MyAccount 페이지 구조 변경 및 CompanyRecruitments로 파일명 변경

- 기존 MyAccount 페이지를 CompanyRecruitments로 이름 변경
- 새로운 MyAccount 페이지 구현을 위해 기존 페이지의 역할 변경

* refactor: visaRegistration 관련 파일 구조 변경

- visaRegistration 페이지를 src/pages/employee에서 src/pages로 이동
- 관련 기능을 src/features/employee/visaRegistration에서 src/features/visaRegistration으로 이동

* feat: Table 컴포넌트 구현

* feat: 변경된 고용주 마이페이지 구현

* refactor: 회사 관련 공통 기능을 features/companies로 이동 및 CompanyInfo 수정

* refactor: CompanyRecruitments 페이지 이름을 MyCompany로 변경

* feat: EmployerMyAccount path 설정

* refactor: 불필요한 코드 삭제 및 폴더명 일관성 있게 변경

* Feat/#50 Select 컴포넌트 구현 (#53)

* feat: add Select component

* feat: add useGlobalSelect and useSelect

* refactor: move directory

* refactor: EmployerMyAccount 페이지에서 mock 데이터 분리 및 코드 정리

* refactor: visaRegistration 및 applicants 페이지의 mock, style 파일 분리

* refactor: RecruitmentList 컴포넌트 리팩토링 및 RecruitmentsTable 분리

* refactor: CompanyInfo 반응형 디자인 수정

* refactor: 외국인 번호 및 비자 발급 일자 등록 페이지 스타일 수정

* refactor: Header 컴포넌트의 닉네임 버튼을 사용자 프로필 이미지로 변경

* Refactor/#54 Modal 컴포넌트 재설계 (#55)

* chore: add loadable component package

* feat: implement modal management system with context and dynamic loading

* refactor: 코드 리뷰 반영

- formValid를 useMemo로 관리
- validateForeignerNumber 함수를 별도 파일로 분리

* Feat/#56 메인 페이지 API 연동 (#57)

* chore: setting mockServiceWorker

* feat: add useFetchRecruitments hooks and recruitmentsMockHandler

* feat: add useFetchSlides hooks and slidesMockHandler

* feat: add Spinner component

* feat: add AsyncBoundary component

* chore: add msw-storybook-addon

* Feat/#58 OAuth 구글 로그인 구현 (#59)

* feat: add useGoogleOAuth hook

* feat: add Loading page

* chore: add MemoryRouter to decorators

* Feat/#60 가입자 정보 선택 API 연동 (#61)

* feat: add useRegister hook

* fix: change role prop value

* style: Button 컴포넌트 Props 이름변경 theme->design

* feat: 근로자마이페이지 아이콘 설정

* feat: 근로자 마이페이지 구현

* feat: 근로자마이페이지 라우터 설정

* style: Button props 이름 변경

* feat: msw 세팅 및 API path 작성

* feat: 구인글 등록 API 연결 및 msw 세팅

* fix: 구인글 업로드 mock 핸들러 수정

* feat: 근로자 마이페이지 mock 핸들러 추가

* feat: 근로자 마이페이지 API 연결 및 msw 설정

* feat: 이력서 페이지 구현 (#63)

- react-hook-form 을 사용했습니다.
- api 명세서에 맞게 이름,주소,번호,경력,자기소개,한국어실력을 필수값으로 받게 했습니다.

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>

* fix: change button prop

* feat: add GitHub Actions workflow for linting and type checking

---------

Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>
Co-authored-by: LEE YONGJIN <129606040+tteokbokki-master@users.noreply.github.com>

* docs: update README.md

* docs: update README.md

* docs: update README.md

* refactor: 지원서 페이지 버튼 수정
- 버튼 스타일을 수정했습니다.

* feat: 지원서 페이지 api 연동(msw)
- recruitmentId를 Path Parameter로 사용했습니다.
- msw로 성공 요청만 받을 수 있게 했습니다.

---------

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>
Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>

* Feat/#69 API 연동 (#80)

* feat: 고용주 마이페이지 API 연결

* feat: 내 회사 페이지 API 연결

* feat: 지원자 목록 페이지 API 연결

* feat: 계약하기 팝업창 API 연결

* feat: 외국인 번호 및 비자 발급 일자 등록 페이지 API 연결

* chore: 의존성 재설치

* chore: 의존성 업데이트

* chore: CI 일관성 확보를 위해 의존성 업데이트

---------

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>

* feat: 근로계약서 path 설정

* Feat/#81 구인 글 상세 화면 수정 및 API 연결 (#83)

* 7, 8주차 산출물 (#70)

* Update README.md

* Refactor/#47 6주차 코드리뷰 리팩토링 (#48)

* refactor: separate style and prop-related constants

* refactor: remove lambda function

* refactor: restructure SignUp and RecruitmentHeader components

* feat: separate Button and Text component of SignIn

* feat: add SignUpText component

* refactor: remove auth page's barrel file

* Feat/#42, #43 지원자 목록 페이지 및 팝업 구현 (#51)

* feat: 지원자 목록 페이지 구현

* feat: 지원자 목록 페이지 스토리북 생성

* refactor: ApplicantList 테이블 컴포넌트 분리

* feat: Applicants path 설정

* feat: 지원자 목록 페이지에 계약 관련 팝업 추가

* fix: 이미지 경로 수정 및 불필요한 태그 제거

* refactor: MyAccount 페이지 구조 변경 및 CompanyRecruitments로 파일명 변경

- 기존 MyAccount 페이지를 CompanyRecruitments로 이름 변경
- 새로운 MyAccount 페이지 구현을 위해 기존 페이지의 역할 변경

* refactor: visaRegistration 관련 파일 구조 변경

- visaRegistration 페이지를 src/pages/employee에서 src/pages로 이동
- 관련 기능을 src/features/employee/visaRegistration에서 src/features/visaRegistration으로 이동

* feat: Table 컴포넌트 구현

* feat: 변경된 고용주 마이페이지 구현

* refactor: 회사 관련 공통 기능을 features/companies로 이동 및 CompanyInfo 수정

* refactor: CompanyRecruitments 페이지 이름을 MyCompany로 변경

* feat: EmployerMyAccount path 설정

* refactor: 불필요한 코드 삭제 및 폴더명 일관성 있게 변경

* Feat/#50 Select 컴포넌트 구현 (#53)

* feat: add Select component

* feat: add useGlobalSelect and useSelect

* refactor: move directory

* refactor: EmployerMyAccount 페이지에서 mock 데이터 분리 및 코드 정리

* refactor: visaRegistration 및 applicants 페이지의 mock, style 파일 분리

* refactor: RecruitmentList 컴포넌트 리팩토링 및 RecruitmentsTable 분리

* refactor: CompanyInfo 반응형 디자인 수정

* refactor: 외국인 번호 및 비자 발급 일자 등록 페이지 스타일 수정

* refactor: Header 컴포넌트의 닉네임 버튼을 사용자 프로필 이미지로 변경

* Refactor/#54 Modal 컴포넌트 재설계 (#55)

* chore: add loadable component package

* feat: implement modal management system with context and dynamic loading

* refactor: 코드 리뷰 반영

- formValid를 useMemo로 관리
- validateForeignerNumber 함수를 별도 파일로 분리

* Feat/#56 메인 페이지 API 연동 (#57)

* chore: setting mockServiceWorker

* feat: add useFetchRecruitments hooks and recruitmentsMockHandler

* feat: add useFetchSlides hooks and slidesMockHandler

* feat: add Spinner component

* feat: add AsyncBoundary component

* chore: add msw-storybook-addon

* Feat/#58 OAuth 구글 로그인 구현 (#59)

* feat: add useGoogleOAuth hook

* feat: add Loading page

* chore: add MemoryRouter to decorators

* Feat/#60 가입자 정보 선택 API 연동 (#61)

* feat: add useRegister hook

* fix: change role prop value

* style: Button 컴포넌트 Props 이름변경 theme->design

* feat: 근로자마이페이지 아이콘 설정

* feat: 근로자 마이페이지 구현

* feat: 근로자마이페이지 라우터 설정

* style: Button props 이름 변경

* feat: msw 세팅 및 API path 작성

* feat: 구인글 등록 API 연결 및 msw 세팅

* fix: 구인글 업로드 mock 핸들러 수정

* feat: 근로자 마이페이지 mock 핸들러 추가

* feat: 근로자 마이페이지 API 연결 및 msw 설정

* feat: 이력서 페이지 구현 (#63)

- react-hook-form 을 사용했습니다.
- api 명세서에 맞게 이름,주소,번호,경력,자기소개,한국어실력을 필수값으로 받게 했습니다.

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>

* fix: change button prop

* feat: add GitHub Actions workflow for linting and type checking

---------

Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>
Co-authored-by: LEE YONGJIN <129606040+tteokbokki-master@users.noreply.github.com>

* docs: update README.md

* docs: update README.md

* docs: update README.md

* refactor: 구인글 상세 정보 반환 페이지 수정
- 데이터 타입을 api 명세에 맞게 변경했습니다.
- recruitmentsDetail 컴포넌트의 스타일을 변경했습니다.
- 타입을 별도로 분리했습니다.

* feat: 구인글 상세화면 api 연동(msw)
- api 명세서에 맞게 요청 로직을 구현했습니다.
- msw를 통해 반환 값을 받을 수 있게 했습니다.

---------

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>
Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>

* Refactor/#71 9주차 기능 개선 및 리팩토링 (#84)

* refactor: 컴포넌트 이름과 일치하도록 파일 이름 변경

* refactor: 고용주 마이페이지에 사인등록 버튼 추가 및 디자인 변경

* feat: 고용주 마이페이지의 사인 등록 버튼 클릭 시 사인 등록 페이지로 이동 기능 추가

* feat: 회사 목록 로딩 중 Spinner 추가 및 위치 조정

* feat: 공고글 목록 로딩 중 Spinner 추가

* refactor: ContractModal의 버튼들을 ModalButtons로 분리

* feat: 계약 팝업창 확인 버튼 클릭 시 근로계약서 페이지로 이동 기능 추가

* Feat/#85 Header 컴포넌트 기능 추가 및 개선 (#87)

* feat: 로그인 여부 및 사용자 타입에 따라 Header 구성 변경

- 로그인 여부에 따라 로그인/로그아웃 버튼 표시
- 고용주인 경우 '채용공고 등록' 버튼 추가
- 로컬스토리지에서 사용자 정보를 가져와 반영

* feat: Header 네비게이션 기능 추가

- 로고 클릭 시 홈 화면으로 이동
- 로그인 버튼 클릭 시 로그인 페이지로 이동
- 채용공고 등록 버튼 클릭 시 구인글 업로드 페이지로 이동
- 프로필 이미지 클릭 시 마이페이지로 이동

* feat: 근로자 마이페이지 path 연결 완료

* feat: 구인글 업로드 path 연결 완료

* feat: 회사 등록 API 연결

* feat: i18n 번역 데이터 - 구인글 업로드, 근로자 마이페이지, 회사 등록 페이지

* refactor: 구인글 업로드 및 근로자 마이페이지 폴더구조 정리

* feat: 고용주 근로계약서 작성 API 연결 및 구인글 작성 API 폴더구조 정리

* feat: 근로자 근로계약서 불러오기 API 연결

* feat: 근로자 근로계약서 저장하기 API 연결

---------

Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>
Co-authored-by: LEE YONGJIN <129606040+tteokbokki-master@users.noreply.github.com>

* Weekly branch merge to Develop branch (#97)

* Refactor/#47 6주차 코드리뷰 리팩토링 (#48)

* refactor: separate style and prop-related constants

* refactor: remove lambda function

* refactor: restructure SignUp and RecruitmentHeader components

* feat: separate Button and Text component of SignIn

* feat: add SignUpText component

* refactor: remove auth page's barrel file

* Feat/#42, #43 지원자 목록 페이지 및 팝업 구현 (#51)

* feat: 지원자 목록 페이지 구현

* feat: 지원자 목록 페이지 스토리북 생성

* refactor: ApplicantList 테이블 컴포넌트 분리

* feat: Applicants path 설정

* feat: 지원자 목록 페이지에 계약 관련 팝업 추가

* fix: 이미지 경로 수정 및 불필요한 태그 제거

* refactor: MyAccount 페이지 구조 변경 및 CompanyRecruitments로 파일명 변경

- 기존 MyAccount 페이지를 CompanyRecruitments로 이름 변경
- 새로운 MyAccount 페이지 구현을 위해 기존 페이지의 역할 변경

* refactor: visaRegistration 관련 파일 구조 변경

- visaRegistration 페이지를 src/pages/employee에서 src/pages로 이동
- 관련 기능을 src/features/employee/visaRegistration에서 src/features/visaRegistration으로 이동

* feat: Table 컴포넌트 구현

* feat: 변경된 고용주 마이페이지 구현

* refactor: 회사 관련 공통 기능을 features/companies로 이동 및 CompanyInfo 수정

* refactor: CompanyRecruitments 페이지 이름을 MyCompany로 변경

* feat: EmployerMyAccount path 설정

* refactor: 불필요한 코드 삭제 및 폴더명 일관성 있게 변경

* Feat/#50 Select 컴포넌트 구현 (#53)

* feat: add Select component

* feat: add useGlobalSelect and useSelect

* refactor: move directory

* refactor: EmployerMyAccount 페이지에서 mock 데이터 분리 및 코드 정리

* refactor: visaRegistration 및 applicants 페이지의 mock, style 파일 분리

* refactor: RecruitmentList 컴포넌트 리팩토링 및 RecruitmentsTable 분리

* refactor: CompanyInfo 반응형 디자인 수정

* refactor: 외국인 번호 및 비자 발급 일자 등록 페이지 스타일 수정

* refactor: Header 컴포넌트의 닉네임 버튼을 사용자 프로필 이미지로 변경

* Refactor/#54 Modal 컴포넌트 재설계 (#55)

* chore: add loadable component package

* feat: implement modal management system with context and dynamic loading

* refactor: 코드 리뷰 반영

- formValid를 useMemo로 관리
- validateForeignerNumber 함수를 별도 파일로 분리

* Feat/#56 메인 페이지 API 연동 (#57)

* chore: setting mockServiceWorker

* feat: add useFetchRecruitments hooks and recruitmentsMockHandler

* feat: add useFetchSlides hooks and slidesMockHandler

* feat: add Spinner component

* feat: add AsyncBoundary component

* chore: add msw-storybook-addon

* Feat/#58 OAuth 구글 로그인 구현 (#59)

* feat: add useGoogleOAuth hook

* feat: add Loading page

* chore: add MemoryRouter to decorators

* Feat/#60 가입자 정보 선택 API 연동 (#61)

* feat: add useRegister hook

* fix: change role prop value

* style: Button 컴포넌트 Props 이름변경 theme->design

* feat: 근로자마이페이지 아이콘 설정

* feat: 근로자 마이페이지 구현

* feat: 근로자마이페이지 라우터 설정

* style: Button props 이름 변경

* feat: msw 세팅 및 API path 작성

* feat: 구인글 등록 API 연결 및 msw 세팅

* fix: 구인글 업로드 mock 핸들러 수정

* feat: 근로자 마이페이지 mock 핸들러 추가

* feat: 근로자 마이페이지 API 연결 및 msw 설정

* feat: 이력서 페이지 구현 (#63)

- react-hook-form 을 사용했습니다.
- api 명세서에 맞게 이름,주소,번호,경력,자기소개,한국어실력을 필수값으로 받게 했습니다.

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>

* fix: change button prop

* feat: add GitHub Actions workflow for linting and type checking

* Feat/#74 사인 등록 페이지 구현 (#75)

* 7, 8주차 산출물 (#70)

* Update README.md

* Refactor/#47 6주차 코드리뷰 리팩토링 (#48)

* refactor: separate style and prop-related constants

* refactor: remove lambda function

* refactor: restructure SignUp and RecruitmentHeader components

* feat: separate Button and Text component of SignIn

* feat: add SignUpText component

* refactor: remove auth page's barrel file

* Feat/#42, #43 지원자 목록 페이지 및 팝업 구현 (#51)

* feat: 지원자 목록 페이지 구현

* feat: 지원자 목록 페이지 스토리북 생성

* refactor: ApplicantList 테이블 컴포넌트 분리

* feat: Applicants path 설정

* feat: 지원자 목록 페이지에 계약 관련 팝업 추가

* fix: 이미지 경로 수정 및 불필요한 태그 제거

* refactor: MyAccount 페이지 구조 변경 및 CompanyRecruitments로 파일명 변경

- 기존 MyAccount 페이지를 CompanyRecruitments로 이름 변경
- 새로운 MyAccount 페이지 구현을 위해 기존 페이지의 역할 변경

* refactor: visaRegistration 관련 파일 구조 변경

- visaRegistration 페이지를 src/pages/employee에서 src/pages로 이동
- 관련 기능을 src/features/employee/visaRegistration에서 src/features/visaRegistration으로 이동

* feat: Table 컴포넌트 구현

* feat: 변경된 고용주 마이페이지 구현

* refactor: 회사 관련 공통 기능을 features/companies로 이동 및 CompanyInfo 수정

* refactor: CompanyRecruitments 페이지 이름을 MyCompany로 변경

* feat: EmployerMyAccount path 설정

* refactor: 불필요한 코드 삭제 및 폴더명 일관성 있게 변경

* Feat/#50 Select 컴포넌트 구현 (#53)

* feat: add Select component

* feat: add useGlobalSelect and useSelect

* refactor: move directory

* refactor: EmployerMyAccount 페이지에서 mock 데이터 분리 및 코드 정리

* refactor: visaRegistration 및 applicants 페이지의 mock, style 파일 분리

* refactor: RecruitmentList 컴포넌트 리팩토링 및 RecruitmentsTable 분리

* refactor: CompanyInfo 반응형 디자인 수정

* refactor: 외국인 번호 및 비자 발급 일자 등록 페이지 스타일 수정

* refactor: Header 컴포넌트의 닉네임 버튼을 사용자 프로필 이미지로 변경

* Refactor/#54 Modal 컴포넌트 재설계 (#55)

* chore: add loadable component package

* feat: implement modal management system with context and dynamic loading

* refactor: 코드 리뷰 반영

- formValid를 useMemo로 관리
- validateForeignerNumber 함수를 별도 파일로 분리

* Feat/#56 메인 페이지 API 연동 (#57)

* chore: setting mockServiceWorker

* feat: add useFetchRecruitments hooks and recruitmentsMockHandler

* feat: add useFetchSlides hooks and slidesMockHandler

* feat: add Spinner component

* feat: add AsyncBoundary component

* chore: add msw-storybook-addon

* Feat/#58 OAuth 구글 로그인 구현 (#59)

* feat: add useGoogleOAuth hook

* feat: add Loading page

* chore: add MemoryRouter to decorators

* Feat/#60 가입자 정보 선택 API 연동 (#61)

* feat: add useRegister hook

* fix: change role prop value

* style: Button 컴포넌트 Props 이름변경 theme->design

* feat: 근로자마이페이지 아이콘 설정

* feat: 근로자 마이페이지 구현

* feat: 근로자마이페이지 라우터 설정

* style: Button props 이름 변경

* feat: msw 세팅 및 API path 작성

* feat: 구인글 등록 API 연결 및 msw 세팅

* fix: 구인글 업로드 mock 핸들러 수정

* feat: 근로자 마이페이지 mock 핸들러 추가

* feat: 근로자 마이페이지 API 연결 및 msw 설정

* feat: 이력서 페이지 구현 (#63)

- react-hook-form 을 사용했습니다.
- api 명세서에 맞게 이름,주소,번호,경력,자기소개,한국어실력을 필수값으로 받게 했습니다.

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>

* fix: change button prop

* feat: add GitHub Actions workflow for linting and type checking

---------

Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>
Co-authored-by: LEE YONGJIN <129606040+tteokbokki-master@users.noreply.github.com>

* docs: update README.md

* docs: update README.md

* docs: update README.md

* chore: react-signature-canvas 설치

* feat: 사인 등록 페이지 구현
- 사인을 이미지를 파일로 만든 후 서버로 전송합니다.
- 요청 로직도 같이 구현했습니다.
- 요청 완료 후 리다이렉트는 로그인 이후 구현

* feat: 사인 등록 페이지 api 연동(msw)

---------

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>
Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>

* Feat/#77 회사 등록 페이지 구현 (#78)

* "refactor: default function 컴포넌트로 변경"

* feat: 회사 등록 페이지 구현

* feat: 회사 등록 페이지 라우터 설정

* style: 회사 이미지 업로드기능 구현

* Feat/#76 지원서 페이지 API 연동(msw) (#79)

* 7, 8주차 산출물 (#70)

* Update README.md

* Refactor/#47 6주차 코드리뷰 리팩토링 (#48)

* refactor: separate style and prop-related constants

* refactor: remove lambda function

* refactor: restructure SignUp and RecruitmentHeader components

* feat: separate Button and Text component of SignIn

* feat: add SignUpText component

* refactor: remove auth page's barrel file

* Feat/#42, #43 지원자 목록 페이지 및 팝업 구현 (#51)

* feat: 지원자 목록 페이지 구현

* feat: 지원자 목록 페이지 스토리북 생성

* refactor: ApplicantList 테이블 컴포넌트 분리

* feat: Applicants path 설정

* feat: 지원자 목록 페이지에 계약 관련 팝업 추가

* fix: 이미지 경로 수정 및 불필요한 태그 제거

* refactor: MyAccount 페이지 구조 변경 및 CompanyRecruitments로 파일명 변경

- 기존 MyAccount 페이지를 CompanyRecruitments로 이름 변경
- 새로운 MyAccount 페이지 구현을 위해 기존 페이지의 역할 변경

* refactor: visaRegistration 관련 파일 구조 변경

- visaRegistration 페이지를 src/pages/employee에서 src/pages로 이동
- 관련 기능을 src/features/employee/visaRegistration에서 src/features/visaRegistration으로 이동

* feat: Table 컴포넌트 구현

* feat: 변경된 고용주 마이페이지 구현

* refactor: 회사 관련 공통 기능을 features/companies로 이동 및 CompanyInfo 수정

* refactor: CompanyRecruitments 페이지 이름을 MyCompany로 변경

* feat: EmployerMyAccount path 설정

* refactor: 불필요한 코드 삭제 및 폴더명 일관성 있게 변경

* Feat/#50 Select 컴포넌트 구현 (#53)

* feat: add Select component

* feat: add useGlobalSelect and useSelect

* refactor: move directory

* refactor: EmployerMyAccount 페이지에서 mock 데이터 분리 및 코드 정리

* refactor: visaRegistration 및 applicants 페이지의 mock, style 파일 분리

* refactor: RecruitmentList 컴포넌트 리팩토링 및 RecruitmentsTable 분리

* refactor: CompanyInfo 반응형 디자인 수정

* refactor: 외국인 번호 및 비자 발급 일자 등록 페이지 스타일 수정

* refactor: Header 컴포넌트의 닉네임 버튼을 사용자 프로필 이미지로 변경

* Refactor/#54 Modal 컴포넌트 재설계 (#55)

* chore: add loadable component package

* feat: implement modal management system with context and dynamic loading

* refactor: 코드 리뷰 반영

- formValid를 useMemo로 관리
- validateForeignerNumber 함수를 별도 파일로 분리

* Feat/#56 메인 페이지 API 연동 (#57)

* chore: setting mockServiceWorker

* feat: add useFetchRecruitments hooks and recruitmentsMockHandler

* feat: add useFetchSlides hooks and slidesMockHandler

* feat: add Spinner component

* feat: add AsyncBoundary component

* chore: add msw-storybook-addon

* Feat/#58 OAuth 구글 로그인 구현 (#59)

* feat: add useGoogleOAuth hook

* feat: add Loading page

* chore: add MemoryRouter to decorators

* Feat/#60 가입자 정보 선택 API 연동 (#61)

* feat: add useRegister hook

* fix: change role prop value

* style: Button 컴포넌트 Props 이름변경 theme->design

* feat: 근로자마이페이지 아이콘 설정

* feat: 근로자 마이페이지 구현

* feat: 근로자마이페이지 라우터 설정

* style: Button props 이름 변경

* feat: msw 세팅 및 API path 작성

* feat: 구인글 등록 API 연결 및 msw 세팅

* fix: 구인글 업로드 mock 핸들러 수정

* feat: 근로자 마이페이지 mock 핸들러 추가

* feat: 근로자 마이페이지 API 연결 및 msw 설정

* feat: 이력서 페이지 구현 (#63)

- react-hook-form 을 사용했습니다.
- api 명세서에 맞게 이름,주소,번호,경력,자기소개,한국어실력을 필수값으로 받게 했습니다.

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>

* fix: change button prop

* feat: add GitHub Actions workflow for linting and type checking

---------

Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>
Co-authored-by: LEE YONGJIN <129606040+tteokbokki-master@users.noreply.github.com>

* docs: update README.md

* docs: update README.md

* docs: update README.md

* refactor: 지원서 페이지 버튼 수정
- 버튼 스타일을 수정했습니다.

* feat: 지원서 페이지 api 연동(msw)
- recruitmentId를 Path Parameter로 사용했습니다.
- msw로 성공 요청만 받을 수 있게 했습니다.

---------

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>
Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>

* Feat/#69 API 연동 (#80)

* feat: 고용주 마이페이지 API 연결

* feat: 내 회사 페이지 API 연결

* feat: 지원자 목록 페이지 API 연결

* feat: 계약하기 팝업창 API 연결

* feat: 외국인 번호 및 비자 발급 일자 등록 페이지 API 연결

* chore: 의존성 재설치

* chore: 의존성 업데이트

* chore: CI 일관성 확보를 위해 의존성 업데이트

---------

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>

* feat: 근로계약서 path 설정

* Feat/#81 구인 글 상세 화면 수정 및 API 연결 (#83)

* 7, 8주차 산출물 (#70)

* Update README.md

* Refactor/#47 6주차 코드리뷰 리팩토링 (#48)

* refactor: separate style and prop-related constants

* refactor: remove lambda function

* refactor: restructure SignUp and RecruitmentHeader components

* feat: separate Button and Text component of SignIn

* feat: add SignUpText component

* refactor: remove auth page's barrel file

* Feat/#42, #43 지원자 목록 페이지 및 팝업 구현 (#51)

* feat: 지원자 목록 페이지 구현

* feat: 지원자 목록 페이지 스토리북 생성

* refactor: ApplicantList 테이블 컴포넌트 분리

* feat: Applicants path 설정

* feat: 지원자 목록 페이지에 계약 관련 팝업 추가

* fix: 이미지 경로 수정 및 불필요한 태그 제거

* refactor: MyAccount 페이지 구조 변경 및 CompanyRecruitments로 파일명 변경

- 기존 MyAccount 페이지를 CompanyRecruitments로 이름 변경
- 새로운 MyAccount 페이지 구현을 위해 기존 페이지의 역할 변경

* refactor: visaRegistration 관련 파일 구조 변경

- visaRegistration 페이지를 src/pages/employee에서 src/pages로 이동
- 관련 기능을 src/features/employee/visaRegistration에서 src/features/visaRegistration으로 이동

* feat: Table 컴포넌트 구현

* feat: 변경된 고용주 마이페이지 구현

* refactor: 회사 관련 공통 기능을 features/companies로 이동 및 CompanyInfo 수정

* refactor: CompanyRecruitments 페이지 이름을 MyCompany로 변경

* feat: EmployerMyAccount path 설정

* refactor: 불필요한 코드 삭제 및 폴더명 일관성 있게 변경

* Feat/#50 Select 컴포넌트 구현 (#53)

* feat: add Select component

* feat: add useGlobalSelect and useSelect

* refactor: move directory

* refactor: EmployerMyAccount 페이지에서 mock 데이터 분리 및 코드 정리

* refactor: visaRegistration 및 applicants 페이지의 mock, style 파일 분리

* refactor: RecruitmentList 컴포넌트 리팩토링 및 RecruitmentsTable 분리

* refactor: CompanyInfo 반응형 디자인 수정

* refactor: 외국인 번호 및 비자 발급 일자 등록 페이지 스타일 수정

* refactor: Header 컴포넌트의 닉네임 버튼을 사용자 프로필 이미지로 변경

* Refactor/#54 Modal 컴포넌트 재설계 (#55)

* chore: add loadable component package

* feat: implement modal management system with context and dynamic loading

* refactor: 코드 리뷰 반영

- formValid를 useMemo로 관리
- validateForeignerNumber 함수를 별도 파일로 분리

* Feat/#56 메인 페이지 API 연동 (#57)

* chore: setting mockServiceWorker

* feat: add useFetchRecruitments hooks and recruitmentsMockHandler

* feat: add useFetchSlides hooks and slidesMockHandler

* feat: add Spinner component

* feat: add AsyncBoundary component

* chore: add msw-storybook-addon

* Feat/#58 OAuth 구글 로그인 구현 (#59)

* feat: add useGoogleOAuth hook

* feat: add Loading page

* chore: add MemoryRouter to decorators

* Feat/#60 가입자 정보 선택 API 연동 (#61)

* feat: add useRegister hook

* fix: change role prop value

* style: Button 컴포넌트 Props 이름변경 theme->design

* feat: 근로자마이페이지 아이콘 설정

* feat: 근로자 마이페이지 구현

* feat: 근로자마이페이지 라우터 설정

* style: Button props 이름 변경

* feat: msw 세팅 및 API path 작성

* feat: 구인글 등록 API 연결 및 msw 세팅

* fix: 구인글 업로드 mock 핸들러 수정

* feat: 근로자 마이페이지 mock 핸들러 추가

* feat: 근로자 마이페이지 API 연결 및 msw 설정

* feat: 이력서 페이지 구현 (#63)

- react-hook-form 을 사용했습니다.
- api 명세서에 맞게 이름,주소,번호,경력,자기소개,한국어실력을 필수값으로 받게 했습니다.

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>

* fix: change button prop

* feat: add GitHub Actions workflow for linting and type checking

---------

Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>
Co-authored-by: LEE YONGJIN <129606040+tteokbokki-master@users.noreply.github.com>

* docs: update README.md

* docs: update README.md

* docs: update README.md

* refactor: 구인글 상세 정보 반환 페이지 수정
- 데이터 타입을 api 명세에 맞게 변경했습니다.
- recruitmentsDetail 컴포넌트의 스타일을 변경했습니다.
- 타입을 별도로 분리했습니다.

* feat: 구인글 상세화면 api 연동(msw)
- api 명세서에 맞게 요청 로직을 구현했습니다.
- msw를 통해 반환 값을 받을 수 있게 했습니다.

---------

Co-authored-by: kangkibong <rkdqudgus9806@gmail.com>
Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>

* Refactor/#71 9주차 기능 개선 및 리팩토링 (#84)

* refactor: 컴포넌트 이름과 일치하도록 파일 이름 변경

* refactor: 고용주 마이페이지에 사인등록 버튼 추가 및 디자인 변경

* feat: 고용주 마이페이지의 사인 등록 버튼 클릭 시 사인 등록 페이지로 이동 기능 추가

* feat: 회사 목록 로딩 중 Spinner 추가 및 위치 조정

* feat: 공고글 목록 로딩 중 Spinner 추가

* refactor: ContractModal의 버튼들을 ModalButtons로 분리

* feat: 계약 팝업창 확인 버튼 클릭 시 근로계약서 페이지로 이동 기능 추가

* Feat/#85 Header 컴포넌트 기능 추가 및 개선 (#87)

* feat: 로그인 여부 및 사용자 타입에 따라 Header 구성 변경

- 로그인 여부에 따라 로그인/로그아웃 버튼 표시
- 고용주인 경우 '채용공고 등록' 버튼 추가
- 로컬스토리지에서 사용자 정보를 가져와 반영

* feat: Header 네비게이션 기능 추가

- 로고 클릭 시 홈 화면으로 이동
- 로그인 버튼 클릭 시 로그인 페이지로 이동
- 채용공고 등록 버튼 클릭 시 구인글 업로드 페이지로 이동
- 프로필 이미지 클릭 시 마이페이지로 이동

* feat: 근로자 마이페이지 path 연결 완료

* feat: 구인글 업로드 path 연결 완료

* feat: 회사 등록 API 연결

* feat: i18n 번역 데이터 - 구인글 업로드, 근로자 마이페이지, 회사 등록 페이지

* refactor: 구인글 업로드 및 근로자 마이페이지 폴더구조 정리

* feat: 고용주 근로계약서 작성 API 연결 및 구인글 작성 API 폴더구조 정리

* feat: 근로자 근로계약서 불러오기 API 연결

* feat: 근로자 근로계약서 저장하기 API 연결

* fix: 이전 merge된 코드에서 발생한 오류 및 잘못된 부분 수정

* chore: npm 패키지 업데이트

---------

Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>
Co-authored-by: YIMSEBIN <jij09123@gmail.com>
Co-authored-by: LEE YONGJIN <129606040+tteokbokki-master@users.noreply.github.com>

---------

Co-authored-by: yimsebin <jij09123@gmail.com>
Co-authored-by: Kim Jian <81233665+KimJi-An@users.noreply.github.com>
Co-authored-by: KimJi-An <j0012a@naver.com>
Co-authored-by: LEE YONGJIN <129606040+tteokbokki-master@users.noreply.github.com>

kang-kibong committed:
refactor: rename VisaRegistrationForm.styles

kang-kibong committed:
Merge remote-tracking branch 'origin/Review' into Master
